js/admin.js

```diff
@@ -134,4 +137,99 @@ jQuery(document).ready(function($) {
     event.preventDefault();
   });
 
+
+  // Handle publication selection
+  $postMediumPublicationIdSelect = $("#post-medium-publication-id-select");
+
+  $("#medium-publication-id .edit-medium-publication-id").click(function () {
+    if ($postMediumPublicationIdSelect.is(":hidden")) {
+      $("#medium-publication-id-hidden").val($postMediumPublicationIdSelect.find("input:radio:checked").val())
+      $postMediumPublicationIdSelect.slideDown("fast", function() {
+        $postMediumPublicationIdSelect.find('input[type="radio"]')
+            .first().focus();
+      });
+      $(this).hide();
+    }
+    return false;
+  });
+
+  $postMediumPublicationIdSelect.find(".save-post-medium-publication-id").click(function(event) {
+    $postMediumPublicationIdSelect.slideUp("fast");
+    $("#medium-publication-id .edit-medium-publication-id").show().focus();
+    $("#post-medium-publication-id-display").html($postMediumPublicationIdSelect.find("input:radio:checked + label").html());
+    var publishable = $postMediumPublicationIdSelect.find('input:radio:checked').data("publishable");
+    $publicStatusRadio = $postMediumStatusSelect.find('input:radio[value="public"]');
+    $unlistedStatusRadio = $postMediumStatusSelect.find('input:radio[value="unlisted"]');
+    if (publishable) {
+      $publicStatusRadio.removeAttr("disabled");
+      $unlistedStatusRadio.removeAttr("disabled");
+    } else {
+      $publicStatusRadio.attr("disabled", "disabled");
+      $unlistedStatusRadio.attr("disabled", "disabled");
+      if ($publicStatusRadio.prop("checked") || $unlistedStatusRadio.prop("checked")) {
+        $postMediumStatusSelect.find('input:radio[value="draft"]').prop("checked", true);
+        $("#post-medium-status-display").html($postMediumStatusSelect.find("input:radio:checked + label").html());
+      }
+    }
+    event.preventDefault();
+  });
+
+  $postMediumPublicationIdSelect.find(".cancel-post-medium-publication-id").click(function(event) {
+    $postMediumPublicationIdSelect.slideUp("fast", function () {
+      $("#medium-publication-id-radio-" + $("#medium-publication-id-hidden").val()).prop("checked", true);
+    });
+    $("#medium-publication-id .edit-medium-publication-id").show().focus();
+    event.preventDefault();
+  });
+
+
+  // Handle refreshing of publications.
+  $refreshPublicationsButton = $("#medium-refresh-publications");
+  $mediumPublicationsDescription = $("#medium-publications-description");
+  $mediumDefaultPublicationIdSelect = $("#medium_default_publication_id");
```

Comment:
because php variables can't have dashes, all the submittable elements use underscores. but for some reason, non submittable elements in wp codebase use dashes, so this matches both styles. i was a :sadpanda: when i wrote this.